doc/sphinx/dev_git_intro.rst

```diff
@@ -52,46 +52,211 @@ Clone a local repository onto your machine
+
+Reverting commits
+^^^^^^^^^^^^^^^^^
+If you want to revert to the commit(s) before you pulled in updates:
+
+1. Find the commit hash(es) with the updates, in your git log
+::
+  git log
+
+or consult the commit log in the web interface
+
+2. Revert each commit in order from newest to oldest
+::
+  git revert <newer commit hash>
+  git revert <older commit hash>
+
+3. Push the updates to the remote branch
+::
+  git push origin feature/<my_feature_name> --force
+
+Updating a branch with a 2-step merge 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If you are concerned with updates breaking your development branch, but don't want to deal with undoing commits, you can test the updates in a copy of your feature branch, then merge the copy branch into your feature branch:
+
+1. Check out your feature branch
+::
+  git checkout feature/<my_feature_name>
+
+2. Check out a new branch from the feature branch
+::
+  git checkout -b <test_branch_name>
+
+3. Merge develop into the test branch using the procedure described in the previous section
 
-   - If you are unfamiliar with git and want to practice with the commands listed here, we recommend you to create an additional feature branch just for this. Remember: your changes will not affect NOAA's repo until you've submitted a pull request through the GitHub webpage and accepted by the lead-team programmer.
+4. Test the branch with the MDTF framework software
 
-   - If you encounter problems during practice, you can first try looking for *plain English* instructions to unmess the situation at `Dangit, Git?! <https://dangitgit.com/>`__.
+5. Check out your feature branch, then merge the test branch into the feature branch
+::
+  git checkout feature/<my_feature_name>
+  git merge <test_branch_name>
 
-- Write your code! A useful command is ``git status`` to remind you what branch you're on and changes you've made (but have not committed yet). ``git branch -a`` lists all branches with ``*`` indicating the branch you're on.
+6. Push the updates to your remote branch
+::
+  git push origin feature/<my_feature_name>
 
-.. (TODO: tests ...)
-.. (TODO: adding files...)
-.. (- Commit changes with ``git commit -m <your commit message>``.) Somehow -m never works for YH.
-.. Good commit messages are key to making the project's history useful. To make this easier, instead of using the ``-m`` flag, you can configure git to launch your text editor of choice with ``git config --global core.editor "<command string to launch your editor>"``.
-.. - To provide further information, add a blank line after the summary and wrap text to 72 columns if your editor supports it (this makes things display nicer on some tools). Here's an `example <https://github.com/NOAA-GFDL/MDTF-diagnostics/commit/225b29f30872b60621a5f1c55a9f75bbcf192e0b>`__.
+7. Delete the test branch
+::
+  git branch -D <test_branch_name>
+
+Updating your feature branch by rebasing it onto the develop branch (preferred method)
```

Comment:
If this is the preferred method, I recommend it goes first. Or maybe I am confused what are alternative methods and what are subsections steps because I'm not seeing section/outline numbering (maybe due to git's rendering of the rst file)?

This is how the file appears to me through the github view:
Start Coding
   steps 1-4
Updating your feature branch by merging in changes from the develop branch
    steps 1-4, then the following two subsections:
    Reverting commits
    Updating a branch with a 2-step merge
          steps 1-6 (note that steps 3 & 4 are more indented than the others)
Updating your feature branch by rebasing it onto the develop branch (preferred method)

So I'm assuming that there are two methods for updating our feature branch (with changes from upstream/develop):

Updating your feature branch by merging in changes from the develop branch
Updating your feature branch by rebasing it onto the develop branch (preferred method)

Reply:
Yes, there are 2 methods to update your feature branch: 
1) the more involved rebase, which groups all of the commits in your branch together ahead of the develop branch commits at the branch tip
2) the simpler merge, which orders all commits on both branches by time

I agree about changing the order.